### PR TITLE
Migrate to OneLocBuild

### PR DESCRIPTION
**Pull request by jonfortescue**
Documentation (pending merge into arcade) can be found [here](https://github.com/jonfortescue/arcade/blob/OneLocBuildDoc/Documentation/OneLocBuild.md).

This depends on arcade latest, so the upcoming PR should be merged. For now, the two new files introduced in arcade latest are included as a workaround (once arcade latest is promoted and taken, there shouldn't be any merge conflicts).

We're still in the process of getting repos onboarded to this, so samples come from test runs. [Here](https://dev.azure.com/dnceng/internal/_build/results?buildId=1044368&view=logs&j=6aa31087-d1b8-5ee1-af29-f0fe63353381) is this working in roslyn-analyzers (code branch [here](https://dev.azure.com/dnceng/internal/_git/dotnet-roslyn-analyzers?version=GBLocalizationTests)). For the alternative method, check a winforms test run [here](https://dev.azure.com/dnceng/internal/_build/results?buildId=1042681&view=logs&j=6aa31087-d1b8-5ee1-af29-f0fe63353381&t=87599fee-7665-5958-325d-e95f1da72bc1) (code branch [here](https://dev.azure.com/dnceng/internal/_git/dotnet-winforms?version=GBLocalizationTests)).

### CODE REVIEW

File: eng/common/generate-locproject.ps1 [hunk 1]
```diff
@@ -0,0 +1,101 @@
+Param(
```
[the remark this answers is not in the document]
Comment (jonfortescue) [reply eng/common/generate-locproject.ps1:1]: Yes precisely, this is part of the most recent arcade update.

**Comment by jonfortescue** [comment]
@sharwell @jmarolf ready to merge!

**Comment by jmarolf** [comment]
@jonfortescue can you verify that this does not break [this](https://dev.azure.com/dnceng/internal/_build?definitionId=830) build pipeline?

**Comment by jonfortescue** [comment]
@jmarolf Looks to be harmless (the job it adds succeeds): https://dev.azure.com/dnceng/internal/_build/results?buildId=1073240&view=results

**Review by jmarolf — APPROVED** [review]

**Review by sharwell — APPROVED** [review]

**Pull request merged**